src/app/profile/my-spaces/my-spaces.component.spec.ts

```diff
@@ -74,161 +70,173 @@ describe('MySpacesComponent', () => {
+            }
+            if (key === 'displaySharedSpaces') {
+              return of([spaceMock1, spaceMock2]);
+            }
+          });
+          mockBroadcaster.broadcast.and.callThrough();
+          return mockBroadcaster;
+        }
+      },
+      {
+        provide: BsModalService,
+        useFactory: (): jasmine.SpyObj<BsModalService> => {
+          const mockBsModalService: jasmine.SpyObj<BsModalService> = createMock(BsModalService);
+          mockBsModalService.show.and.returnValue(mockModalRef);
+          return mockBsModalService;
+        }
+      },
+      {
+        provide: SpaceService,
+        useFactory: (): jasmine.SpyObj<SpaceService> => {
+          const mockSpaceService: any = createMock(SpaceService);
+          mockSpaceService.getSpacesByUser.and.returnValue(of([spaceMock1, spaceMock2]));
+          mockSpaceService.deleteSpace = () => {};
+          return mockSpaceService;
+        }
+      },
+      {
+        provide: UserService,
+        useFactory: (): jasmine.SpyObj<UserService> => {
+          const mockUserService: jasmine.SpyObj<UserService> = createMock(UserService);
+          mockUserService.loggedInUser = of(mockUser) as ConnectableObservable<User> & jasmine.Spy;
+          return mockUserService;
+        }
+      },
+      {
+        provide: AuthenticationService,
+        useFactory: (): jasmine.SpyObj<AuthenticationService> => {
+          const mockAuthenticationService: any = createMock(AuthenticationService);
+          mockAuthenticationService.getGitHubToken = () => {};
+          mockAuthenticationService.getToken.and.returnValue('mock-token');
+          return mockAuthenticationService;
+        }
+      },
+      {
+        provide: ErrorHandler,
+        useFactory: (): jasmine.SpyObj<ErrorHandler> => {
+          const mockErrorHandler: jasmine.SpyObj<ErrorHandler> = createMock(ErrorHandler);
+          mockErrorHandler.handleError.and.callThrough();
+          return mockErrorHandler;
+        }
+      },
+      { provide: WIT_API_URL, useValue: 'https://example.com/api/' }
+    ],
+    schemas: [NO_ERRORS_SCHEMA]
+  });
 
-  describe('#spaces', () => {
-    it('should return the contents of _spaces', () => {
-      component.appliedFilters = [];
-      component.ngOnInit();
-      component.applyFilters(); // initialize _spaces first using applyFilters()
-      let result = component.spaces;
+  describe('#spaces', (): void => {
+    it('should return the contents of _spaces', (): void => {
+      let result: Space[] = testContext.testedDirective.spaces;
       expect(result).toEqual([spaceMock1, spaceMock2]);
     });
   });
 
+  describe('#handlePinChange', (): void => {
+    it('should delegate to savePins and updateSpaces if the selected space exists in the user\'s spaces', (): void => {
+      const component: MySpacesComponent = testContext.testedDirective;
+      spyOn(component, 'savePins').and.callFake((): void => {});
```

Comment:
`callFake((): void => {})` looks like it could just be `stub()` for this purpose.